File: app_manager/appmanager/src/main/java/app/editors/manager/mvp/models/base/Capabilities.java
package app.editors.manager.mvp.models.base;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.List;

public class Capabilities {

    public static final String KEY_LDAP = "ldapEnabled";
    public static final String KEY_SSO_URL = "ssoUrl";
    public static final String KEY_SSO_LABEL = "ssoLabel";
    public static final String KEY_PROVIDERS = "providers";

    @SerializedName(KEY_LDAP)
    @Expose
    private boolean ldapEnabled = false;

    @SerializedName(KEY_SSO_URL)
    @Expose
    private String ssoUrl = "";

    @SerializedName(KEY_SSO_LABEL)
    @Expose
    private String ssoLabel = "";

    @SerializedName(KEY_PROVIDERS)
    @Expose
    private List<String> providers;

    public boolean getLdapEnabled() {
        return ldapEnabled;
    }

    public void setLdapEnabled(boolean ldapEnabled) {
        this.ldapEnabled = ldapEnabled;
    }

    public String getSsoUrl() {
        return ssoUrl;
    }

    public void setSsoUrl(String ssoUrl) {
        this.ssoUrl = ssoUrl;
    }

    public String getSsoLabel() {
        return ssoLabel;
    }

    public void setSsoLabel(String ssoLabel) {
        this.ssoLabel = ssoLabel;
    }

    public List<String> getProviders() {
        return providers;
    }

    public void setProviders(List<String> providers) {
        this.providers = providers;
    }
}

File: app_manager/core/src/main/java/app/documents/core/network/manager/models/base/Capabilities.kt
package app.documents.core.network.manager.models.base

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class Capabilities {

    @SerializedName(KEY_LDAP)
    @Expose
    var ldapEnabled = false

    @SerializedName(KEY_SSO_URL)
    @Expose
    var ssoUrl = ""

    @SerializedName(KEY_SSO_LABEL)
    @Expose
    var ssoLabel = ""

    @SerializedName(KEY_PROVIDERS)
    @Expose
    var providers: List<String>? = null

    companion object {
        const val KEY_LDAP = "ldapEnabled"
        const val KEY_SSO_URL = "ssoUrl"
        const val KEY_SSO_LABEL = "ssoLabel"
        const val KEY_PROVIDERS = "providers"
    }
}